lib/pure/collections/sequtils.nim

```diff
@@ -675,16 +668,16 @@ template mapIt*(s, op: untyped): untyped =
     block:
       var it{.inject.}: type(items(s));
       op))
-  var result: seq[outType]
   when compiles(s.len):
-    evalOnce(t, s)
+    # using t_D20180808T211500 to avoid https://github.com/nim-lang/Nim/issues/8580
+    evalOnce(t_D20180808T211500, s, type(s) isnot openArray)
```

Comment:
This is nasty. Using a `block:` is a somewhat cleaner workaround.

Reply:
Instead of checking whether `s` is an `openArray` I'd just add another `compiles((let _ = s))` to the `when` above.

Reply:
This usage of ``compiles`` seems to be a hack...  :-/

Reply:
Yep, but I can't think of a better way to shunt the bare `openArray` literals into the other code path

Reply:
> This is nasty. Using a block: is a somewhat cleaner workaround.

used a block; much cleaner, thanks!

> compiles((let _ = s))

had thought of that but will likely lead to other bugs and hide other issues

> Yep, but I can't think of a better way to shunt the bare openArray literals into the other code path

not sure what you're suggesting besides `compiles` ; what could go wrong with `type(s) isnot openArray` ?

Reply:
as discussed here https://gitter.im/nim-lang/Nim?at=5b6c0cf3637ee66082c775e8, I changed to `compiles((let _ = s))` (see comments in code)